src/frontend/js/components/SearchFilterValueParent/index.tsx

```diff
@@ -54,24 +54,27 @@ export const SearchFilterValueParent = ({ filter, value }: SearchFilterValuePare
   const showChildren = userShowChildren !== null ? !!userShowChildren : hasActiveChildren;
 
   const [children, setChildren] = useState([] as FilterValue[]);
-  useAsyncEffect(async () => {
-    if (showChildren) {
-      // Get only the filters & facet counts for the children of the current parent
-      const childrenResponse = await fetchList('courses', {
-        ...courseSearchParams,
-        [`${filter.name}_include`]: childrenPathMatch,
-        scope: 'filters',
-      });
+  const { data: childrenResponse } = useCourseSearch(
```

Comment:
IMO the best outcome for this refactor would be to use the query state to render different things between loading/success/failure, and ditch the `[children, setChildren]` state altogether. 

I don't think it makes sense to store react-query return values in component state. Then you can also remove the whole `useEffect`call  below.

Reply:
You are totally right. I removed the state and process date through a `useMemo` instead of `useEffect`

Reply:
I don't understand why we need a `useMemo` here: as far as I understand it, it's supposed to be a performance optimisation tool, but we're not doing any heavy work.

Error reporting could probably move to `useCourseSearch` directly (maybe in an `onError` handler?), then we're basically just setting a default value `const children = childrenResponse?.content.filters[filter.name].values || []`

Reply:
Ok, I removed the `useMemo` and use the `onSettled` method to handle error to be keep the detailed error message from `SearchFilterValueParent`.